lib/files.js

```diff
@@ -36,11 +46,13 @@ var putPart = function(info, callback) {
     if (!info.json) {
         return callback();
     }
-    var file = path.join(options.dir, info.json.name, info.version, 'index.json');
-    process.nextTick(function() {
-        mkdirp(path.dirname(file), function() {
-            process.nextTick(function() {
-                fs.writeFile(file, JSON.stringify(info.json, null, 4) + '\n', callback);
+    hooks.versionJson(info, callback, function() {
```

Comment:
Same as the `tarball` hook, missing the arguments and the action to stop the write.